Prepare for 0.7 release

ludwigschwardt opened the pull request:
More than I thought...

Ignoring the massive but cosmetic flake8 changes, the only other changes involve test cleanups. I silence the rcond FutureWarning that polluted the test output, as well as a custom condition warning (but I also check that the latter happens as expected). I also had to relax one test to make it pass on NumPy 1.15.0.

ludwigschwardt committed:
Update copyright year to 2018

ludwigschwardt committed:
More flake8

A very tedious line length cleanup (blame OCD).

ludwigschwardt committed:
Silence rcond FutureWarning in numpy >= 1.14.0

ludwigschwardt committed:
Ignore __pycache__

ludwigschwardt committed:
Catch and verify condition warning in test

This cleans up the test output and also checks that the poor
conditioning is picked up as expected.

ludwigschwardt committed:
Slightly relax test precision (blame numpy 1.15)

NumPy 1.15.0 introduced a change that tweaked the output of the test
ever so slightly. Instead of the expected [-0.1, 0.2, -0.3, 0, 0.5]
we get [-0.1000000, 0.1999999, -0.2999993, -0.000001804571, 0.5000011]).
The last value seems to have one fewer decimal precision than before.
No time to dig down on this further...

ludwigschwardt committed:
Bump version number to 0.7 and update changelog

ludwigschwardt committed:
Add trove classifiers and python_requires

This highlights that both Python 2.7 and 3.x are supported.

bmerry commented on scikits/fitting/__init__.py:
```diff
@@ -68,7 +70,7 @@
 ----------------
 
-- :func:`squash` : Flatten array, but not necessarily all the way to a 1-D array
+- :func:`squash` : Flatten array, but not necessarily all the way to 1-D array
```
I preferred the way this was.

ludwigschwardt replied:
Fixed.

bmerry commented on scikits/fitting/rbf.py:
```diff
@@ -32,7 +32,7 @@
 
 class RbfScatterFit(ScatterFit):
-    """Do radial basis function (RBF) interpolation of scattered multi-dimensional data.
+    """Do radial basis function (RBF) interpolation of scattered multi-dim data.
```
I preferred the original.

ludwigschwardt replied:
Fixed.

bmerry commented on scikits/fitting/tests/test_linlstsq.py:
```diff
@@ -76,7 +76,7 @@ def test_vs_numpy(self):
         interp = LinearLeastSquaresFit()
         interp.fit(self.x, self.y)
-        params = np.linalg.lstsq(self.x.T, self.y)[0]
+        params = np.linalg.lstsq(self.x.T, self.y, rcond=None)[0]
```
This unfortunately gives an error (`TypeError: a float is required`) in numpy < 0.14. So if you do this, you'll need to specify a minimum of 0.14 in setup.py.

ludwigschwardt replied:
I'll do `rcond=-1` instead, which should be backwards compatible. This only matters for one test anyway.

bmerry commented on NEWS.rst:
```diff
@@ -1,6 +1,12 @@
+----------------
+
+* Python 3 support
+* Clean up tests and more flake8 (line lengths)
```
Add note about fixes for newer numpy?

ludwigschwardt replied:
Since it only affects one test, it is not that special.

ludwigschwardt committed:
Use older version of rcond to support numpy<1.14

Revert to the old default behaviour of rcond because NumPy older than
1.14 still expects the parameter to be a float. It only affects a
single test with well-behaved data that is impervious to this change
in behaviour anyway. This maintains support for at least numpy==1.8.2
and scipy==0.14.1 (did not check the pre-wheel days :-)).

Also fix the rest of PR snags.
